### PR TITLE
Support publishing v2j batches in the event publisher sidecar.

### PR DESCRIPTION
**Pull request by aearly**

### CODE REVIEW

File: baseplate/sidecars/event_publisher.py [hunk 1]
```diff
@@ -181,7 +192,7 @@ def publish_events() -> None:
         {
             "collector": {
                 "hostname": config.String,
-                "version": config.Optional(config.Integer, default=1),
+                "version": config.Optional(config.String, default="2"),
```
Comment (aearly) [inline baseplate/sidecars/event_publisher.py:195]: This is a small functional change with the default value, but it seemed the default value would always cause an error (perhaps that is intended?).

File: baseplate/sidecars/event_publisher.py [hunk 2]
```diff
@@ -86,10 +86,21 @@ def reset(self) -> None:
         self._size = len(self._header) + len(self._end)
 
 
+class V2JBatch(V2Batch):
```
Comment (spladug) [inline baseplate/sidecars/event_publisher.py:89]: could you add a test here please? https://github.com/reddit/baseplate.py/blob/develop/tests/unit/lib/events/publisher_tests.py#L40

just something simple to exercise this code and make sure it doesn't regress in the future

**Review by spladug — CHANGES_REQUESTED** [review]
LGTM modulo a test. thanks!

File: baseplate/sidecars/event_publisher.py (hunk 1 above)
Comment (spladug) [reply baseplate/sidecars/event_publisher.py:195]: oops, a bad default from the days when v1 still existed. this looks good, thanks.

**Review by spladug — APPROVED** [review]
LGTM!

**Pull request merged**